packages/runtime-core/src/componentOptions.ts

```diff
@@ -745,19 +741,6 @@ export function applyOptions(
     })
   }
 
-  // asset options.
-  // To reduce memory usage, only components with mixins or extends will have
-  // resolved asset registry attached to instance.
-  if (asMixin) {
-    resolveInstanceAssets(instance, options, COMPONENTS)
-    resolveInstanceAssets(instance, options, DIRECTIVES)
-    if (__COMPAT__ && isCompatEnabled(DeprecationTypes.FILTERS, instance)) {
-      resolveInstanceAssets(instance, options, FILTERS)
-    }
-
-    if (inheritAttrs !== undefined) instance.inheritAttrs = inheritAttrs
```

Comment:
Note: should only set this if the base component doesn't have the option (which should have highest priority)